src/items/functions.md

```diff
@@ -244,12 +244,36 @@ fn test_only() {
 > Note: Except for lints, it is idiomatic to only use outer attributes on
 > function items.
 
-The attributes that have meaning on a function are [`cfg`], [`deprecated`],
+The attributes that have meaning on a function are [`cfg`], [`cfg_attr`], [`deprecated`],
 [`doc`], [`export_name`], [`link_section`], [`no_mangle`], [the lint check
 attributes], [`must_use`], [the procedural macro attributes], [the testing
 attributes], and [the optimization hint attributes]. Functions also accept
 attributes macros.
 
+## Attributes on function parameters
+
+[Outer attributes][attributes] are allowed on function parameters and the
+permitted [built-in attributes] are restricted to `cfg`, `cfg_attr`, `allow`,
+`warn`, `deny`, and `forbid`.
+
+```rust
+fn len(
+    #[cfg(windows)] slice: &[u16],
+    #[cfg(not(windows))] slice: &[u8],
+) -> usize {
+    slice.len()
+}
+```
+
+Inert helper attributes used by procedural macro attributes applied to items are also
+allowed but be careful to not include these inert attributes in your final `TokenStream`.
+
+```rust
+#[hi_i_am_a_proc_macro_attribute]
+fn foo_oof(#[hello_i_am_an_inert_attribute] arg: u8) {
```

Comment:
Maybe add a comment explaining what this is trying to illustrate?  Maybe something like this:

```rust
// This inert attribute is not formally defined anywhere. The
// `hi_i_am_a_proc_macro_attribute` proc macro is responsible for detecting
// its presence and removing it from the output token stream.
```